api/envoy/config/route/v3/route_components.proto

```diff
@@ -909,13 +923,16 @@ message RouteAction {
     // for additional documentation.
     WeightedCluster weighted_clusters = 3;
 
-    // [#not-implemented-hide:]
-    // Name of the cluster specifier plugin to use to determine the cluster for
-    // requests on this route. The plugin name must be defined in the associated
-    // :ref:`envoy_v3_api_field_config.route.v3.RouteConfiguration.cluster_specifier_plugins`
-    // in the
-    // :ref:`envoy_v3_api_field_config.core.v3.TypedExtensionConfig.name` field.
+    // Name of the cluster provider to use to determine the cluster for requests on this route.
+    // The cluster provider name must be defined in the associated
+    // :ref:`cluster specifier plugins <envoy_v3_api_field_config.route.v3.RouteConfiguration.cluster_specifier_plugins>`
+    // in the :ref:`name <envoy_v3_api_field_config.core.v3.TypedExtensionConfig.name>` field.
     string cluster_specifier_plugin = 37;
+
+    // Custom cluster specifier plugin configuration to use to determine the cluster for requests
+    // on this route.
+    ClusterSpecifierPlugin cluster_plugin = 39
```

Comment:
Suggest calling this something like `cluster_specifier_plugin_config` or maybe `inline_cluster_specifier_plugin`.

If we'd known we were going to add this field originally, I would have suggested calling the existing field `cluster_specifier_plugin_name`, so that we could use `cluster_specifier_plugin` for this new field, but unfortunately, it's too late to change that.

Reply:
`inline_cluster_specifier_plugin` is great.